### ISSUE TITLE
Only last value is kept when parsing PAGP DAD

### ISSUE THREAD

**Issue by FlavienJP** [CiscoTestAutomation/genieparser#794]
Hello,

While working on checks on my infrastructure with Genie, I saw that the parser for cli command "show stackwise-virtual dual-active-detection pagp" with mulitples Port-Channels is resetting the dict [channel_group]  (might be renamed to channel_group**s**] ?).

The faulty line is [there].(https://github.com/CiscoTestAutomation/genieparser/blob/b901c3faeb4d0ebc8d1e1039a5d0be8cd9bbe770/src/genie/libs/parser/iosxe/show_stackwise.py#L235
)  

```
        # Original
        if match9:
            pagp_dad_obj['channel_group'] = {}  # Resetting at each port channel status (multiples LAGs) - Only last one is kept.
            channel_group = int(match9.group("channel_group"))
            pagp_dad_obj['channel_group'][channel_group] = {}
            pagp_dad_obj['channel_group'][channel_group]['port'] = {}

```

I've made a quick and dirty fix but I guess you will find a better way to do it.

```
        # Modified
        if match9:
                channel_group = int(match9.group("channel_group"))
                try :
                    pagp_dad_obj['channel_group'][channel_group] = {}   # Check if group exists
                except :
                    pagp_dad_obj['channel_group'] = {}  # Else create it
                    pagp_dad_obj['channel_group'][channel_group] = {}                
                pagp_dad_obj['channel_group'][channel_group]['port'] = {}
```

Regards,

**Comment by kameswararaot** [comment]
Hi FlavienJP,

I am working on this issue. Could you please share the device output to find out the issue?

**Comment by kameswararaot** [comment]
Hi FlavienJP,

Could you please share the device output to find out the issue?

**Comment by FlavienJP** [comment]
Hello @kameswararaot,

Sorry for the delay I was ooo.

This is an example from my setup :

```
Switch1#show stackwise-virtual dual-active-detection pagp
Load for five secs: 1%/0%; one minute: 1%; five minutes: 1%
Time source is NTP, 19:58:16.384 CET Mon Nov 13 2023
Pagp dual-active detection enabled: Yes
In dual-active recovery mode: No
Recovery Reload: Enabled


Channel group 11
          Dual-Active     Partner              Partner       Partner
Port      Detect Capable  Name                 Port          Version
Twe1/0/1  Yes             Switch2.lan Te1/1/1       1.1
Twe2/0/1  Yes             Switch2.lan Te2/1/1       1.1

Channel group 12
          Dual-Active     Partner              Partner       Partner
Port      Detect Capable  Name                 Port          Version
Twe1/0/2  No              Switch3.lan                 None          N/A
Twe2/0/2  No              Switch3.lan                 None          N/A

Channel group 13
          Dual-Active     Partner              Partner       Partner
Port      Detect Capable  Name                 Port          Version
Twe1/0/3  No              Switch4.lan Te1/1/1       N/A
Twe2/0/3  No              Switch4.lan Te2/1/1       N/A

Channel group 14
          Dual-Active     Partner              Partner       Partner
Port      Detect Capable  Name                 Port          Version
Twe1/0/4  No              Switch5.lan Te1/1/1       N/A
Twe2/0/4  No              Switch4.lan Te2/1/1       N/A
```

We can have multiple channel groups on a switch and one (or more) can have PaGP DPD enabled, the code is currently retaining only the last channel group instead of all in a dict.

Thanks,

**Comment by FlavienJP** [comment]
Hello @kameswararaot 

Thanks for your correction & update, can you attach the PR id ? Or will it part of the December release 23.12 ?

Regards,

**Comment by kameswararaot** [comment]
Hi FlavienJP,

PR has been merged. It will reflect in next release. Hence I am closing this ticket. If you are still experiencing the same problem in next release also, please let me know, and I will reopen the ticket.

**Comment by FlavienJP** [comment]
Hello @kameswararaot,

Any idea of when the next release will be available ? 2023.12 has been missed.

Regards,
